### PR TITLE
Feature/refactor movies

### PR DESCRIPTION
**Pull request by johnpooler**
TODO--->  ratings DTO READ

### CODE REVIEW

File: app/models/ReleasedMovieWithID.scala [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package models
+
+import java.time.LocalDateTime
+
+import akka.http.javadsl.model.DateTime
+import play.api.libs.json.OFormat
+import reactivemongo.bson.BSONObjectID
+
+case class ReleasedMovieWithID(
+                      var _id: BSONObjectID,
```
Comment (RafSobolQAC) [inline app/models/ReleasedMovieWithID.scala]: Codacy Issue found: [Field name '_id' does not match the regular expression '^[a-z][A-Za-z0-9]*$'](https://app.codacy.com/manual/RafSobolQAC/cinema-team-one/pullRequest?prid=5277123)

**Comment by RafSobolQAC** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Clones removed
==============
+ app/services/UpcomingServices.scala  -1
+ app/services/ReleasedServices.scala  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/RafSobolQAC/cinema-team-one/pullRequest?prid=5277123&bid=17281019)

File: app/models/UpcomingMovieWithID.scala [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package models
+
+import java.time.LocalDateTime
+
+import akka.http.javadsl.model.DateTime
+import play.api.libs.json.OFormat
+import reactivemongo.bson.BSONObjectID
+
+case class UpcomingMovieWithID(
+                      var _id: BSONObjectID,
```
Comment (RafSobolQAC) [inline app/models/UpcomingMovieWithID.scala]: Codacy Issue found: [Field name '_id' does not match the regular expression '^[a-z][A-Za-z0-9]*$'](https://app.codacy.com/manual/RafSobolQAC/cinema-team-one/pullRequest?prid=5277123)

**Review by malikbensalem — APPROVED** [review]
Looks Good

File: app/controllers/UpcomingController.scala [hunk 3]
```diff
@@ -1,16 +1,13 @@
 package controllers
 
 import javax.inject.Inject
-import models.MovieWithID
+import models.UpcomingMovieWithID
```
Comment (nathan-rees) [inline app/controllers/UpcomingController.scala:4]: gg trash

**Review by nathan-rees — APPROVED** [review]

**Comment by RafSobolQAC** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           

Clones added
============
- test/controllers/ReleasedControllerTest.scala  2
         

Clones removed
==============
+ app/services/ReleasedServices.scala  -1
+ app/services/UpcomingServices.scala  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/RafSobolQAC/cinema-team-one/pullRequest?prid=5277123&bid=17281019)

File: app/services/ReleasedServices.scala [hunk 4]
```diff
@@ -1,7 +1,8 @@
 package services
 
-import javax.inject.Inject
-import models.{Movie, MovieWithID}
+import javax.inject.{Inject, Singleton}
+import models.MovieWithID
```
Comment (RafSobolQAC) [inline app/services/ReleasedServices.scala]: Codacy Issue found: [Use brackets to group imports from the same package](https://app.codacy.com/manual/RafSobolQAC/cinema-team-one/pullRequest?prid=5277123)

File: app/controllers/ReleasedController.scala [hunk 5]
```diff
@@ -1,10 +1,12 @@
 package controllers
 
 import javax.inject.Inject
+import models.ReleasedMovieWithID
```
Comment (RafSobolQAC) [inline app/controllers/ReleasedController.scala]: Codacy Issue found: [Use brackets to group imports from the same package](https://app.codacy.com/manual/RafSobolQAC/cinema-team-one/pullRequest?prid=5277123)

File: app/controllers/ReleasedController.scala [hunk 6]
```diff
@@ -20,18 +22,23 @@ class ReleasedController @Inject()(
...
     releasedServices.getMovies.map { movies =>
-
-
-      Ok(views.html.movieInfo(movies.filter(movie => id == movie._id.toString()).head))
+      movies.map(movie => movie.title -> movie.screenings)
+    }.map { movies =>
+      Ok(movies.toString())
     }
   }
+  def releasedMovieInfo(id: String) = Action.async { implicit request: Request[AnyContent] =>
```
Comment (RafSobolQAC) [inline app/controllers/ReleasedController.scala]: Codacy Issue found: [The method takes a value that is controlled by the client](https://app.codacy.com/manual/RafSobolQAC/cinema-team-one/pullRequest?prid=5277123)

**Comment by RafSobolQAC** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Clones added
============
- test/controllers/ReleasedControllerTest.scala  2
         

Clones removed
==============
+ app/services/UpcomingServices.scala  -1
+ app/services/ReleasedServices.scala  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/RafSobolQAC/cinema-team-one/pullRequest?prid=5277123&bid=17281019)

File: app/controllers/ReleasedController.scala [hunk 7]
```diff
@@ -20,18 +21,21 @@ class ReleasedController @Inject()(
...
     releasedServices.getMovies.map { movies =>
-
-
-      Ok(views.html.movieInfo(movies.filter(movie => id == movie._id.toString()).head))
+      movies.map(movie => movie.title -> movie.screenings)
+    }.map { movies =>
+      Ok(movies.toString())
+    }
+  }
+  def releasedMovieInfo(id: String) = Action.async { implicit request: Request[AnyContent] =>
```
Comment (RafSobolQAC) [inline app/controllers/ReleasedController.scala:40]: Codacy Issue found: [The method takes a value that is controlled by the client](https://app.codacy.com/manual/RafSobolQAC/cinema-team-one/pullRequest?prid=5277123)

**Pull request merged**